### PR TITLE
Adding LICENSE/NOTICE/DISCLAIMER to NBMs, packing OSGi jars into NBMs…

### PR DESCRIPTION
**Pull request by jlahoda**
… when producing update center from the main checkout, so that it holds the appropriate licenses.

If we'd like to release the update center content, it needs to follow the rules - i.e. have LICENCE/NOTICE/DISCLAIMER, etc.

### CODE REVIEW

**Comment by geertjanw** [comment]
Looks good and makes sense to me.

File: nbbuild/antsrc/org/netbeans/nbbuild/MakeNBM.java [hunk 1]
```diff
@@ -1029,6 +1053,29 @@ private Document createInfoXml(final Attributes attr) throws BuildException {
...
+        }
+
+        log("Creating fake info.xml for OSGi bundle", Project.MSG_VERBOSE);
+        
+        String pub = "-//NetBeans//DTD Autoupdate Module Info 2.5//EN";
+        String sys = "http://www.netbeans.org/dtds/autoupdate-info-2_5.dtd";
+        Document doc = domimpl.createDocument(null, "module", domimpl.createDocumentType("module", pub, sys));
+        
+        try (JarFile jf = new JarFile(osgiJar)) {
+            MakeUpdateDesc.fakeOSGiInfoXml(jf, osgiJar, doc);
```
Comment (matthiasblaesing) [inline nbbuild/antsrc/org/netbeans/nbbuild/MakeNBM.java]: I noticed a difference between regular modules and OSGI(fake) modules: In the normal modules a license section is present in the `info.xml`. The LICENSE is present in the generated `nbm`. So from that perspective this should be good - I'm just wondering whether we need to cover this.

File: nbbuild/antsrc/org/netbeans/nbbuild/extlibs/CreateLicenseSummary.java [hunk 2]
```diff
@@ -289,6 +296,18 @@ private void evaluateBinaries(final PrintWriter licenseWriter, final PrintWriter
...
+        if (moduleFiles != null) {
+            for (Resource r : moduleFiles) {
+                try (InputStream is = r.getInputStream()) {
+                    long crc = computeCRC32(is);
+                    Map<String, String> headers = crc2License.get(crc);
+                    if (headers != null) {
+                        binaries2LicenseHeaders.put(r.getName(), headers);
+                    }
+                }
+                System.err.println(r.as(File.class));
```
Comment (matthiasblaesing) [inline nbbuild/antsrc/org/netbeans/nbbuild/extlibs/CreateLicenseSummary.java]: `System.err` see other locations

File: nbbuild/templates/common.xml [hunk 3]
```diff
@@ -427,7 +427,11 @@
         <fileset dir="${cluster}" id="module.executable.files" includes="${nbm.executable.files}"/>
     </target>
 
-    <target name="nbm" depends="init,netbeans,-nbm-prompt-for-storepass,-init-executables" description="Build NBM archive.">
+    <target name="-init-extra.nbm.files" unless="extra.nbm.files.provided"> <!-- fallback -->
+        <zipfileset dir="${cluster}" id="extra.nbm.files" includes="[NOTHING]"/>
```
Comment (matthiasblaesing) [inline nbbuild/templates/common.xml]: I don't understand the "[NOTHING]" here - what is the intention?

File: nbbuild/antsrc/org/netbeans/nbbuild/MakeNBM.java [hunk 4]
```diff
@@ -581,25 +594,32 @@ public void execute () throws BuildException {
...
-            copy.setProject(getProject());
-            copy.setOwningTarget(getOwningTarget());
-            copy.setFile(module);
-            copy.setTofile(new File(nbm.getAbsolutePath().replaceFirst("[.]nbm$", ".jar")));
-            copy.execute();
-            // XXX possibly sign it
-            // XXX could try to run pack200, though not if it was signed
-            return;
+            System.err.println("attr=" + attr);
+            System.err.println("alwaysCreateNBM=" + alwaysCreateNBM);
```
[the remark this answers is not in the document]
Comment (jlahoda) [reply nbbuild/antsrc/org/netbeans/nbbuild/MakeNBM.java]: Oops, right, I must have overseen those when looking at the patch. These should be removed now.

File: nbbuild/templates/common.xml (hunk 3 above)
Comment (jlahoda) [reply nbbuild/templates/common.xml]: The goal here is to ensure there is a zipfileset "extra.nbm.files" defined, but is empty. There is a similar thing nearby (-init-executables), so I simply copied the includes here. common.xml is used by platform applications as well, so we should not be enforcing core NB constraints here, so I tried to make the changes compatibly.

Comment (matthiasblaesing) [reply nbbuild/templates/common.xml]: Thanks - that way it makes sense.

**Pull request merged**